File: b/Disk.py
from enum import Enum

DiskType = Enum('SSD', 'HDD')

class Disk():
	__type: Enum
	__capacity: int
	__price: int

	def __init__(self, type:DiskType, capacity:int, price:int):
		self.__type = type
		self.__capacity = capacity
		self.__price = price

	def setType(self, type:DiskType): self.__type = type
	def setCapacity(self, capacity:int): self.__capacity = capacity
	def setPrice(self, price:int): self.__price = price

	def getType(self): return self.__type
	def getCapacity(self): return self.__capacity
	def getPrice(self): return self.__price

	def printDisk(self):
		print("Tipe Disk : ", self.__type)
		print("Kapasitas Disk : ", self.__capacity)
		print("Harga Disk : ", self.__price)

File: a/Disk.cpp
enum DiskType{SSD, HDD};

class Disk
{
	private:
		DiskType type;
		int capacity;
		int price;
	
	public:
		// construct
		Disk(DiskType type, int capacity, int price)
		{
			this->type = type;
			this->capacity = capacity;
			this->price = price;
		}

		// default construct
		Disk(){}

		void setType(DiskType type) { this->type = type; }
		void setCapacity(int capacity) { this->capacity = capacity; }
		void setPrice(int price) { this->price = price; }

		DiskType getType() { return this->type; }
		int getCapacity() { return this->capacity; }
		int getPrice() { return this->price; }

		void printDisk()
		{
			switch (this->type)
			{
			case 0:
				cout << "Tipe Disk : SSD" << endl;
				break;
			
			case 1:
				cout << "Tipe Disk : HDD" << endl;
				break;
			}
			cout << "Kapasitas Disk : " << this->capacity << endl;
			cout << "Harga Disk : " << this->price << endl;
		}

		// deconstruct
		~Disk(){}
};